src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -68,8 +68,9 @@ def __init__(
         fuse_model: bool = True,
         dtype: Optional[torch.dtype] = None,
     ):
-        self.device = device or next(model.parameters()).device
-        self.model = model.to(self.device)
+        model_device: str = next(model.parameters()).device.type
```

Comment:
FYI: We now (3.2) have nice `infer_model_device(m:nn.Module)` and `infer_model_dtype(m:nn.Module)` methods that does `next(model.parameters()).device.type` and a bit more checks under the hood. So you may want to use these instead

Reply:
Good point, I changed it here 797e50685f43ad51d0c0f17b7b96d79edfc3eee8